src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -34,43 +37,72 @@
      * @return the command based on the user input
      * @throws ParseException if the user input does not conform the expected format
      */
-    public Command parseCommand(String userInput) throws ParseException {
+    public Command parseCommand(String userInput, CommandBox commandBox) throws ParseException {
         final Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
-        if (!matcher.matches()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
-        }
+        final Matcher dynamicMatcher = DYNAMIC_COMMAND_FORMAT.matcher(userInput.trim());
+
+        if (dynamicMatcher.matches() && dynamicMatcher.group("inputType").equals("dynamic")) {
+            final String arguments = dynamicMatcher.group("arguments").trim();
+            String[] splitArgs = arguments.split(" ");
+
+            if (isCommand(arguments, splitArgs)) {
+                commandBox.setDynamicAsFalse();
+                return new ListCommand();
+            } else {
+                commandBox.setDynamicAsTrue();
+                return new FindCommandParser().parse(arguments);
+            }
+        } else {
+            if (!matcher.matches()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+            }
+
+            final String commandWord = matcher.group("commandWord");
+            final String arguments = matcher.group("arguments");
 
-        final String commandWord = matcher.group("commandWord");
-        final String arguments = matcher.group("arguments");
-        switch (commandWord) {
+            switch (commandWord) {
```

Comment:
The indentation of the `switch` block does not comply with the coding standard

Reply:
May I ask which part I should change for this? My IDE could not detect any check style errors for this part.

Reply:
Sorry, I didn't notice that the whole block is indented. This is correct.